### PR TITLE
Add Curator and Spark example in docs

### PR DESCRIPTION
**Pull request by ronjer30**
## Description
Updated docs to include example to read and write datasets with NeMo Curator and Apache Spark
## Usage
N/A
## Checklist

- [X] I am familiar with the [Contributing Guide](https://github.com/NVIDIA/NeMo-Curator/blob/main/CONTRIBUTING.md).
- N/A [ ] New or Existing tests cover these changes.
- [X] The documentation is up to date with these changes.

### CODE REVIEW

File: docs/user-guide/sparkother.rst [hunk 1]
```diff
@@ -0,0 +1,98 @@
+.. _data-curator-sparkother:
+
+==========================================================================
+Reading and writing datasets with NeMo Curator and Apache Spark
+==========================================================================
+-----------------------------------------
+Background
+-----------------------------------------
+
+NeMo Curator uses the ``DocumentDataset`` to read and write ``JSONL`` and ``Parquet`` files.
```
Comment (sarahyurick) [inline docs/user-guide/sparkother.rst:10]:
```suggestion
NeMo Curator uses the ``DocumentDataset`` class to read and write JSONL and Parquet files. 
```

File: docs/user-guide/sparkother.rst [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+==========================================================================
+Reading and writing datasets with NeMo Curator and Apache Spark
+==========================================================================
+-----------------------------------------
+Background
+-----------------------------------------
+
+NeMo Curator uses the ``DocumentDataset`` to read and write ``JSONL`` and ``Parquet`` files. 
+It is a wrapper around a `Dask DataFrame <https://docs.dask.org/en/stable/dataframe.html>`_.  Apache Spark can read and write JSONL and Parquet files generated by NeMo Curator, and similarly, NeMo Curator can work with the outputs generated by Spark.
```
Comment (sarahyurick) [inline docs/user-guide/sparkother.rst:11]:
```suggestion
It is a wrapper around a `Dask (or Dask-cuDF) DataFrame <https://docs.dask.org/en/stable/dataframe.html>`_.  Apache Spark can read and write JSONL and Parquet files generated by NeMo Curator, and similarly, NeMo Curator can work with the outputs generated by Spark.
```

File: docs/user-guide/sparkother.rst [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+This will create two JSONL files in the directory ``tiny_stories/``:
+
+::
+
+    tiny_stories/
+        0.part
+        1.part
+
+Apache Spark can read these files using standard APIs. Assuming you already have a Spark session running, you can read files in the directory using:
```
Comment (sarahyurick) [inline docs/user-guide/sparkother.rst:52]:
```suggestion
Apache Spark can read these files using standard APIs. Let's first create a Spark session called ``NeMoCuratorExample``, then we can read files in the directory using:
```

Nit, but it looks like you already include the code to start a Spark session below.

File: docs/user-guide/sparkother.rst [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+    # Calculate Word Count
+    stories_df = stories_df.withColumn("WordCount", size(split(stories_df["text"], r'\s+')))
+
+    # Calculate Character Count
+    stories_df = stories_df.withColumn("CharacterCount", length(stories_df["text"]))
+
+    stories_df.write.mode("overwrite").parquet("tiny_stories_transformed")
+
+To interoperate between NeMo Curator ``DocumentDataset`` and Spark DataFrames, we recommend using parquet files for data exchange.
```
Comment (sarahyurick) [inline docs/user-guide/sparkother.rst:78]:
```suggestion
To interoperate between NeMo Curator ``DocumentDataset`` and Spark DataFrames, we recommend using Parquet files for data exchange.  
```

File: docs/user-guide/sparkother.rst [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    stories_df.show()
+
+Let's go ahead and add a couple of columns to the Spark DataFrame: 
+
+.. code-block:: python
+
+    from pyspark.sql.functions import size, split, length
+
+    # Calculate Word Count
+    stories_df = stories_df.withColumn("WordCount", size(split(stories_df["text"], r'\s+')))
```
Comment (sarahyurick) [inline docs/user-guide/sparkother.rst:71]:
```suggestion
    stories_df = stories_df.withColumn("WordCount", size(split(stories_df["text"], r"\s+")))
```

File: docs/user-guide/sparkother.rst [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+    stories_df.write.mode("overwrite").parquet("tiny_stories_transformed")
+
+To interoperate between NeMo Curator ``DocumentDataset`` and Spark DataFrames, we recommend using parquet files for data exchange.  
+The following code snippet demonstrates how to read output from a Spark DataFrame into a NeMo Curator ``DocumentDataset``:
+
+.. code-block:: python
+
+    from nemo_curator.utils.file_utils import get_all_files_paths_under
+
+    #ignores checksum and marker files created by Spark
```
Comment (sarahyurick) [inline docs/user-guide/sparkother.rst:85]:
```suggestion
    # Ignores checksum and marker files created by Spark 
```

File: docs/user-guide/sparkother.rst [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+The following code snippet demonstrates how to read output from a Spark DataFrame into a NeMo Curator ``DocumentDataset``:
+
+.. code-block:: python
+
+    from nemo_curator.utils.file_utils import get_all_files_paths_under
+
+    #ignores checksum and marker files created by Spark 
+    processed_files = [
+         filename for filename in get_all_files_paths_under("tiny_stories_transformed")
+         if not (filename.endswith('.crc') or filename.endswith('_SUCCESS'))
```
Comment (sarahyurick) [inline docs/user-guide/sparkother.rst:88]:
```suggestion
         if not (filename.endswith(".crc") or filename.endswith("_SUCCESS"))
```

File: docs/user-guide/sparkother.rst [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+    ]
+
+    stories_dataset = DocumentDataset.read_parquet(processed_files, backend='pandas')
```
Comment (sarahyurick) [inline docs/user-guide/sparkother.rst:91]:
```suggestion
    stories_dataset = DocumentDataset.read_parquet(processed_files, backend="pandas")
```

**Review by sarahyurick — CHANGES_REQUESTED** [review]
Awesome tutorial and very easy to follow! Added a couple nits but otherwise looks good.

**Comment by ronjer30** [comment]
Thanks for the feedback @sarahyurick, I've addressed all the changes.

**Comment by sarahyurick** [comment]
Looks like you need to fix whitespace issues somewhere, otherwise looks good.

**Comment by ronjer30** [comment]
Thanks @sarahyurick, fixed trailing whitespace

**Comment by sarahyurick** [comment]
Looks like there's still more. They could be extra spaces at the end of some lines.

**Comment by ronjer30** [comment]
Thanks @sarahyurick, pushed another commit, hopefully this resolves all of it.

**Review by sarahyurick — APPROVED** [review]

**Pull request merged**